service/dap/server.go

```diff
@@ -1943,6 +1947,24 @@ func (s *Server) doCommand(command string, asyncSetupDone chan struct{}) {
 		default:
 			stopped.Body.Reason = "breakpoint"
 		}
+		if state.CurrentThread != nil && state.CurrentThread.Breakpoint != nil {
+			// If we are stopped at a logpoint, we want to send an OutputEvent
+			// and continue running.
+			if state.CurrentThread.Breakpoint.Tracepoint {
+				if msg, ok := s.logPoints[state.CurrentThread.Breakpoint.ID]; ok {
+					s.handleLogPoint(msg)
+				}
+				if command == api.Continue {
+					s.doCommand(api.Continue, nil)
```

Comment:
I think we need more thought  and discussion here.
- s.doCommand calling s.doCommand ... can potentially create unlimited stack growth. Maybe 'goto' to the beginning of the current doCommand?
- should we worry about potential race (new request arriving in between and interfering)?

Wonder if a separate worker goroutine for async request would make this easier.

handleRequest:
  For disconnect, pause, terminate, restart, and breakpoint/thread while program is running:
     handle it as done in https://github.com/go-delve/delve/blob/c10b222bad3970ecff635cd33a82c33c13593f39/service/dap/server.go#L381-L460
  For other cases
     try to send to the worker goroutine through a blocking channel.
     if the request can be sent, good.
     if the request can't be sent:
        sendErrorResponse because the target is running.
        
   The worker goroutine listens on a blocking channel until it's closed. The worker goroutine will handle at most one request at a time. When handling this log point case or requests that involve multiple invocations of debugger APIs, the worker goroutine will be occupied and not accept any more requests. What do you think?   

cc @polinasok

Reply:
This was something I wanted to bring up to you guys actually. I was realizing that both log points and next while nexting will require us to resume execution from doCommand, which will require us to synchronize between doCommand and handleRequest.

We also will need to handle the possibility that the user halted the program manually when we hit a log point, which means we won't want to resume execution.